helix-term/src/ui/spinner.rs

```diff
@@ -46,22 +64,31 @@ impl Spinner {
     }
 
     pub fn dots(interval: u64) -> Self {
-        Self::new(vec!["⣾", "⣽", "⣻", "⢿", "⡿", "⣟", "⣯", "⣷"], interval)
+        Self::new(
+            BRAILLE_SPINNER_STRINGS
+                .into_iter()
+                .map(String::from)
+                .collect(),
+            interval,
+        )
     }
 
     pub fn start(&mut self) {
         self.start = Some(Instant::now());
     }
 
     pub fn frame(&self) -> Option<&str> {
-        let idx = (self
-            .start
-            .map(|time| Instant::now().duration_since(time))?
-            .as_millis()
-            / self.interval as u128) as usize
-            % self.count;
-
-        self.frames.get(idx).copied()
+        let idx = if self.count > 1 {
```

Comment:
This adds complexity. Revert the changes and make the spinner in config an Option<Spinner>. If an empty set is passed just use the default.

Reply:
Alternatively: You could make get() always return &str and simply unwrap_or("")

Reply:
The count is necessary since if we have a single frame we don't worry about the call to (and calculations with) `Instant::now()` since the spinner is either on with frame 0 or off. This is independent of the actual frames, so I'm not quite sure how checking an `Option` would work here.